package/piechart.py

```diff
@@ -1,12 +1,14 @@
 import os
+import shutil
 
 
 def plot(snippet_name):
     snippet_path = os.path.join(
         os.path.dirname(__file__), "stash", f"{snippet_name}.py"
     )
     output_path = os.path.join(os.path.dirname(__file__), f"{snippet_name}.py")
-    with open(snippet_path, "r") as file:
-        source_code = file.read()
-    with open(output_path, "w") as output_file:
-        output_file.write(source_code)
+
+    if os.path.isfile(snippet_path):
+        shutil.copyfile(snippet_path, output_path)
+    else:
+        print("Syntax Error!")
```

Comment:
You don't have to check for `isfile` if you're using `shutil.copyfile` it will raise an `FileNotFoundError` so you can avoid that and put the copyfile statement in a try expect block. more of something like this.
```suggestion
   try:
        base_dir = os.path.dirname(__file__)
        snippet_path = os.path.join(base_dir, "stash", f"{snippet_name}.py")
        output_path = os.path.join(base_dir, f"{snippet_name}.py")
        
        shutil.copyfile(snippet_path, output_path)
     # Note you can be more precise here tho 
    except Exception as e:
        print(f"Error: {e}")
```


I'm not , you have to check if this works for existing and not existing files. Thanks